en/02_Developer_Guides/05_Extending/05_Injector.md

```diff
@@ -1,42 +1,42 @@
 ---
-title: Injector
+title: Dependency Injection
 summary: Introduction to using Dependency Injection within Silverstripe CMS.
 icon: code
 ---
 
-# Injector
+# Dependency Injection
 
-The [Injector](api:SilverStripe\Core\Injector\Injector) class is the central manager of inter-class dependencies in Silverstripe CMS. It offers developers the 
-ability to declare the dependencies a class type has, or to change the nature of the dependencies defined by other 
-developers. 
+> ...dependency injection is a design pattern in which an object or function receives other objects or functions that it depends on
+[Wikipedia](https://en.wikipedia.org/wiki/Dependency_injection)
 
+In Silverstripe a combination of the [Injector API](#Injector) and the [Configuration API](../configuration) provide a comprehensive dependency injection pattern. 
 Some of the goals of dependency injection are:
 
 * Simplified instantiation of objects
 * Providing a uniform way of declaring and managing inter-object dependencies
-* Making class dependencies configurable
-* Simplifying the process of overriding or replacing core behaviour
-* Improve testability of code
 * Promoting abstraction of logic
 
-The following sums up the simplest usage of the `Injector` it creates a new object of type `MyClassName` through `create`
+In practical terms it allows developers to: 
 
+* Make class dependencies configurable rather than hard-coded
+* Override or replace core behaviour without needing to alter core code
+* Write more testable code
 
-```php
-use SilverStripe\Core\Injector\Injector;
 
-$object = Injector::inst()->create('MyClassName');
-```
+# Injector
 
-The benefit of constructing objects through this syntax is `ClassName` can be swapped out using the 
-[Configuration API](../configuration) by developers.
+The [Injector](api:SilverStripe\Core\Injector\Injector) class is the central manager of inter-class dependencies in Silverstripe CMS. It offers developers the 
+ability to declare the dependencies a class type has, or to change the nature of the dependencies defined by other 
+developers. 
 
-**app/_config/app.yml**
+## Basic usage
 
-```yml
-SilverStripe\Core\Injector\Injector:
-  MyClassName:
-    class: MyBetterClassName
+The following snippet shows `Injector` creating a new object of type `MyClassName` through its `create` method:
+
+```php
+use SilverStripe\Core\Injector\Injector;
+
+$object = Injector::inst()->create('MyClassName');
```

Comment:
How about using the `::class` notation throughout in the code examples?

```suggestion
use App\MyClassName;
use SilverStripe\Core\Injector\Injector;

$object = Injector::inst()->create(MyClassName::class);
```

Reply:
yeah good point. I guess it doesn't have to be a class but using `::class` does feel more like a 'real' use case. I could use that and make sure there is a reference to being able to use non-class strings.

Reply:
In the yaml config examples it would have to be the string version of the fully namespaced class name, so we can keep it there. I think it's also good to use a namespaced class in the yaml example, too, as people often forget to use the namespaces there.

Reply:
yeah all good - should now be using that syntax 👍